recipe/meta.yaml

```diff
@@ -0,0 +1,151 @@
+{% set name = "torchaudio" %}
+{% set version = "2.1.2" %}
+{% set run_unittests = False %}
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://github.com/pytorch/audio/archive/refs/tags/v{{ version }}.tar.gz
+  sha256: 82c2745a73172b495327ec36c6970ad5ad9d5d2ac44feeaea2617152f9393bf7
+
+build:
+  number: 0
+  skip: True  # [py>311]
+  script:
+    - export USE_FFMPEG=ON                         # [not win]
+    - set USE_FFMPEG=ON                            # [win]
+    - export BUILD_SOX=OFF                         # [not win]
+    - set BUILD_SOX=OFF                            # [win]
+    - export USE_OPENMP=ON                         # [not win]
+    - set USE_OPENMP=ON                            # [win]
+    - export BUILD_TORCHAUDIO_PYTHON_EXTENSION=ON  # [not win]
+    - set BUILD_TORCHAUDIO_PYTHON_EXTENSION=ON     # [win]
+    - export FFMPEG_ROOT=${PREFIX}                 # [not win]
+    - set FFMPEG_ROOT=%LIBRARY_PREFIX%             # [win]
+    - {{ PYTHON }} -m pip install . -vv --no-deps --no-build-isolation
+  missing_dso_whitelist:
+    - "**/libc10.so"                # [linux]
+    - "**/libtorch_cpu.so"          # [linux]
+    - "**/libtorch_python.so"       # [linux]
+    - "**/libtorchaudio_ffmpeg.so"  # [linux or osx]
+    - "**/libtorchaudio.so"         # [linux or osx]
+    - "**/ld64.so.1"                # [s390x]
+    - "**/libc10.dylib"             # [osx]
+    - "**/libtorch_cpu.dylib"       # [osx]
+    - "**/libtorch_python.dylib"    # [osx]
+    - "**/torch_cpu.dll"            # [win]
+    - "**/c10.dll"                  # [win]
+    # because of pytorch --> _openmp_mutex --> libgomp
+    - "**/lib/libgomp.so.*"         # [linux]
```

Comment:
This is bad. We need to explicitly depend on omp on linux here (when not building with mkl variant). You explicitly enable `USE_OPENMP`, so we have a direct dependency on OpenMP.

Reply:
what do you suggest? disabling `USE_OPENMP`?

Reply:
Any suggestion @JeanChristopheMorinPerso ? I'm not really sure what the best course of action would be here.

Reply:
including openmp in the requirements explicitly

Reply:
you mean adding it to run @danpetry?

Reply:
it should be in host and run. If there's a run_export in the openmp recipe then you can just add it to host and it will be added to run, taking care of the DSO check that's failing

Reply:
But libraries like this need to be in host (the environment in which the package is compiled) too, because they provide the header files required by consuming libraries, and also because the linker resolves all symbols at compile-time as well. (the above DSO check is additional to this and checks the run requirements specifically).

Reply:
wait, are you sure it's needed in host though? I don't see info to point to that this is needed in host. The overlinking error is mentioning `_openmp_mutex` being missing in run so I'm tempted to put it in run only.

Reply:
an openmp implementation needs to be in the host and run environment. sorry I was talking about the openmp library rather than the mutex package. I think the mutex can just go in run and libgomp (the openmp implementation which the mutex relates to) is put into host by some other means (probably via the gcc package)

Reply:
it just pulls in libgomp

Reply:
it looks like it's linking against libgomp rather than intel-openmp

Reply:
testing what happens if we ignore_run_exports: libgomp, in a separate PR

Reply:
I don't think `ignore_run_exports: libgomp` would do anything because `libgomp` is not in the `run_export` of any of the dependencies of `_openmp_mutex`.

I believe the reason why we're linking against `libgomp.so` is that we have `pytorch` in host, which in turn depends on `_openmp_mutex` ([it's in its `run` section](https://conda-metadata-app.streamlit.app/?q=pkgs%2Fmain%2Flinux-64%2Fpytorch-2.2.0-cpu_py39hdc00b08_0.conda)), which does nothing but providing `libgomp.so`. We should probably fix `_openmp_mutex`, because clearly something is off with the fact that it's packaging the same as `libgomp`. Also, the fact that `_openmp_mutex` [is in the `run_export` section](https://conda-metadata-app.streamlit.app/?q=pkgs%2Fmain%2Flinux-64%2Flibgomp-11.2.0-h1234567_1.conda) of `libgomp` is wrong in my opinion.

We could try to put `libgomp` both in the `run` and `run_export` section of `pytorch` .

In any case, these fixes go out of the scope of this PR, and I believe for now an acceptable workaround would be to put `_openmp_mutex` in run (like I did a few commits ago).

@JeanChristopheMorinPerso feel free to chime in. For this is a good compromise.

Reply:
@JeanChristopheMorinPerso I decided to follow @danpetry's strategy of **disabling OPENMP** on `linux` altogether due to these issues around `pytroch`, `libgomp` and `_openmp_mutex`

Reply:
>I believe the reason why we're linking against libgomp.so is that we have pytorch in host, which in turn depends on _openmp_mutex ([it's in its run section](https://conda-metadata-app.streamlit.app/?q=pkgs%2Fmain%2Flinux-64%2Fpytorch-2.2.0-cpu_py39hdc00b08_0.conda)), which does nothing but providing libgomp.so.

No. tochaudio is linked ot libgomp because cmake does it. Also, `_openmp_mutex` doesn't provide libgomp. It only contains a symlink `libgomp.so.1` which points to `libgomp.so.1.0.0` contained in the `libgomp` package. See https://github.com/AnacondaRecipes/gcc_toolchain-feedstock/blob/main/recipe/install_scripts/install-openmp_impl.sh#L8.

>We should probably fix _openmp_mutex, because clearly something is off with the fact that it's packaging the same as libgomp.

Like said above, it doesn't package libgomp.

Reply:
It pulls in libgomp as a run requirement though - https://github.com/AnacondaRecipes/gcc_toolchain-feedstock/blob/525581cbf9edd348209e0bf8c56be267e86ee969/recipe/meta.yaml#L373
It doesn't "package" it but basically its purpose seems to be to pull it in and provide that symlink rather than doing any mutual exclusion. Anyway, I think we all seem to agree that the openmp stuff is pretty broken